src/DynamoCore/Extensions/IExtension.cs

```diff
@@ -50,5 +52,20 @@ public interface IExtension : IDisposable
         /// Action to be invoked when shutdown has begun.
         /// </summary>
         void Shutdown();
+
+        /// <summary>
+        /// Scan the PackagesDirectory for packages and attempt to load all of them.
+        /// </summary>
+        /// <param name="preferences">IPreferences instance, which contains 
+        /// a list of packages used by the Package Manager to determine, 
+        /// which packages are marked for deletion.</param>
+        /// <param name="pathManager">IPathManager instance 
+        /// with set of directories paths needed for Package Manager</param>
+        void Load(IPreferences preferences, IPathManager pathManager);
+
+        /// <summary>
+        /// Event which is fired, when it needs to load an assembly with nodes
+        /// </summary>
+        event Action<Assembly> RequestLoadNodeLibrary;
```

Comment:
Hi @pboyer, having this `RequestLoadNodeLibrary` on a seemingly generic interface `IExtension` looks a little odd, after all this is specific to package loading and is only relevant to `PackageManagerExtension`. Are we looking at tweaking this further before the interface gets frozen? I'm thinking of introducing `IDynamoModel` (`DynamoModel` derives from it) and pass that to each `IExtension` during start-up. Then `PackageManagerExtension` can call an interface method `IDynamoModel.LoadNodeFromAssembly` to get things loaded into the library server. Just my 0.0000002 cents.